sdk/cosmos/azure-cosmos/azure/cosmos/aio/_cosmos_client_connection_async.py

```diff
@@ -3052,27 +3052,40 @@ def GetPartitionResolver(self, database_link: str) -> Optional[RangePartitionRes
     # Adds the partition key to options
     async def _AddPartitionKey(self, collection_link, document, options):
         collection_link = base.TrimBeginningAndEndingSlashes(collection_link)
+        new_options = dict(options)
```

Comment:
correct me if I'm wrong, but are we not repeating logic here if we have retries made for the container recreate scenario?

Reply:
This is retrying the preprocessing we do before making a request instead of retrying a request. Here we actually refresh the cache before a request is made. This is only if we have to extract the partition key from a document body, otherwise passing in the partition key when creating/replacing/upserting an item will go through the retry policy.